app/public/js/components/header/settingsNav.jsx

```diff
@@ -34,21 +34,40 @@ class SettingsApp extends Component {
         super(props);
 
         this.state = {
-            isVisible: false
+            isVisible: false,
+            isMouseIn: false
         };
+
+        //Close when we click on the UI
+        window.addEventListener('click', this.closeOut.bind(this), false);
     }
 
+    stopClose (e) {
+        e.stopPropagation();
```

Comment:
can you remove this abstraction please? just use `e.stopPropagation();` where you need to